src/VisualStudio/Core/Impl/CodeModel/AbstractCodeModelService.AbstractNodeLocator.cs

```diff
@@ -21,24 +23,24 @@ protected AbstractNodeLocator(AbstractCodeModelService codeModelService)
 
             protected abstract EnvDTE.vsCMPart DefaultPart { get; }
 
-            protected abstract VirtualTreePoint? GetStartPoint(SourceText text, SyntaxNode node, EnvDTE.vsCMPart part);
-            protected abstract VirtualTreePoint? GetEndPoint(SourceText text, SyntaxNode node, EnvDTE.vsCMPart part);
+            protected abstract VirtualTreePoint? GetStartPoint(SourceText text, OptionSet options, SyntaxNode node, EnvDTE.vsCMPart part);
+            protected abstract VirtualTreePoint? GetEndPoint(SourceText text, OptionSet options, SyntaxNode node, EnvDTE.vsCMPart part);
 
-            protected int GetTabSize(SourceText text)
+            protected int GetTabSize(OptionSet options)
             {
-                return _codeModelService.GetTabSize(text);
+                return options.GetOption(FormattingOptions.TabSize, _codeModelService.Language);
```

Comment:
This is abstract because there is a C# version and a VB version.  We could just get the language from that, and not have `_codeModelService` at all.